src/test/java/io/jenkins/plugins/eggplant/EggplantRunnerBuilderTest.java

```diff
@@ -20,7 +21,8 @@ public void testBuild() throws Exception {
         FreeStyleProject project = jenkins.createFreeStyleProject();
         EggplantRunnerBuilder builder = new EggplantRunnerBuilder();
         builder.setServerURL("http://localhost:8080");
-        builder.setTestConfigId("test-Config-Id");
+        TestConfigId testConfigId = new TestConfigId("test-Config-Id");
+        builder.setTestConfig(testConfigId);
```

Comment:
For the Unit test, by having the getCommand method splitting to smaller pieces. Basically we can perform unit test on each functionality of the method in more precisely.  This will increase our code coverage for unit test.
    @Test
    public void testBuild_testConfigName_validation() throws Exception {

        EggplantRunnerBuilder builder = new EggplantRunnerBuilder();
        builder.setTestConfigName("test-Config-Name");
        BuilderException exception = assertThrows(BuilderException.class, ()-> builder.getBackwardCompatibilityCommands());
        String expectedMessage = "testConfigName found, suiteName or modelName is required.";
        String actualMessage = exception.getMessage();    
        assertTrue(actualMessage.contains(expectedMessage));  
    }

Reply:
> 

Mainly i did so because it was a FreestyleProject instance, thus i thought from UI perspective and set the testConfig instance instead of setting the testConfigId directly. 

I was thinking to add another set of test for Pipeline Project if feasible, that will set the testConfigId directly. Not sure if this sounds sensible to you?

I can remove if this is not needed.